packages/component/src/SendBox/UploadButton.tsx

```diff
@@ -131,13 +151,15 @@ const UploadButton: FC<UploadButtonProps> = ({ className }) => {
         onChange={disabled ? undefined : handleFileChange}
         onClick={disabled ? PREVENT_DEFAULT_HANDLER : undefined}
         readOnly={disabled}
-        ref={inputRef}
+        ref={uploadButtonRef}
         role="button"
         tabIndex={-1}
         type="file"
       />
       <IconButton alt={uploadFileString} aria-label={uploadFileString} disabled={disabled} onClick={handleClick}>
         <AttachmentIcon />
+        {/* When a file is attached, overlay the check icon */}
+        {uploadButtonRef.current?.value && <CheckIcon />}
```

Comment:
Using `uploadButtonRef.current?.value` to check if a file is pending upload, is probably the reason why you need to set `uploadButtonRef.current.value` to `null` after send.

It seems `useFiles` hook was added much later and not really used.